client/src/components/Achievements/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+import { useStateContext } from "../../context";
+import { AcheiveItem } from './AcheiveItem';
+
+export const Achievements = () => {
+  const [campaignsData, setCampaignsData] = useState<any>(null);
+  const { address, contract } = useStateContext();
+
+  const donatedCampaigns = !campaignsData ? [] : campaignsData.filter((campaign: any) => campaign.donators.some((donator: string) => donator === address));
+  const donationCount = donatedCampaigns.length < 1 ? [] : donatedCampaigns.map((campaign: any) => campaign.donators.filter((donator: string) => donator === address)).flat();
+
+  const acheiveList = donationCount.length > 0 ? [
+    {
+      id: 0,
+      title: 'First donate',
+      description: 'make you first donate',
+      success: donationCount.length >= 1,
+    },
+    {
+      id: 1,
+      title: 'Second donate',
+      description: 'make you second donate',
+      success: donationCount.length >= 2,
+    },
+    {
+      id: 2,
+      title: 'Third donate',
+      description: 'make you third donate',
+      success: donationCount.length >= 3,
+    },
+    {
+      id: 3,
+      title: 'Fourth donate',
+      description: 'make you fourth donate',
+      success: donationCount.length >= 4,
+    },
+  ] : [];
```

Comment:
If **donationCount** is empty, you can put it instead of a new empty array.

```suggestion
  ] : donationCount;
```